Dockerfile

```diff
@@ -35,7 +35,7 @@ RUN apk add --no-cache --update alpine-sdk \
 # If you change this value, please also update:
 # /dev.Dockerfile
 # /.github/workflows/main.yml
-FROM golang:1.22.3-alpine as golangbuilder
+FROM golang:1.22.6-alpine as golangbuilder
```

Comment:
Any reason not to add integrity pinning here and other "FROM golang" areas in the code?
PR quick adds this in lnd: https://github.com/lightningnetwork/lnd/pull/9115/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557

Reply:
How does this work across different build architectures that have different images? E.g. building on MacOS vs. building on Linux?

Reply:
Ah, I see, there's kind of like a "root" hash that's for all the architectures. Okay, I'll add that, once everyone has confirmed the reproducible build.

Reply:
A deliverable addressed here:
>Docker-Image digests exist under a cohesive index digest which account for the fact that different architectures will obtain different image files. See additional commentary: https://github.com/docker/hub-feedback/issues/1925#issuecomment-565646445

https://github.com/lightningnetwork/lnd/pull/9115#:~:text=Docker%2DImage%20digests%20exist%20under%20a%20cohesive%20index%20digest%20which%20account%20for%20the%20fact%20that%20different%20architectures%20will%20obtain%20different%20image%20files.%20See%20additional%20commentary%3A%20docker/hub%2Dfeedback%231925%20(comment)